linkedList/linkedList.js

```diff
@@ -0,0 +1,190 @@
+class ListNode {
+    constructor(value) {
+        this.value = value;
+        this.next = null;
+    }
+}
+
+
+class LinkedList {
+    constructor() {
+        this.head = null;
+        this.tail = null;
+        this.length = 0;
+    }
+
+    //Inserting methods
+    addAtHead(value) {
+        if (this.isEmpty()) {
+            this.head = new ListNode(value);
+            this.tail = this.head;
+            this.length++;
+            return this.length;
+        }
+        let node = new ListNode(value);
+        node.next = this.head;
+        this.head = node;
+        return this.length++
+
+    }
+
+    addAtTail(value) {
+        if (this.isEmpty()) {
+            return this.addAtHead(value);
+
+        }
+        const node = new ListNode(value);
+        this.tail.next = node;
+        this.tail = node;
+        this.length++;
+        return this.length;
+
+    }
+    addAtIndex(index, value) {
+        if (index > this.length) {
+            return;
+        } else if (index == this.length) {
+            return this.addAtTail(value);
+        } else if (index <= 0) {
+            return this.addAtHead(value);
+        } else {
+            let node = new ListNode(value);
+            let currentNode , previousNode ;
+           [previousNode, currentNode] = this.getPrevAndCurrNodesAtIndex(index);
+            previousNode.next = node;
+            node.next = currentNode;
+            return this.length++;
+        }
+    }
+    // get methods
+    get(index) {
+        if (index < 0 || index > this.length - 1) {
+            return -1;
+        }
+        let currentNode = this.head;
+        while (index) {
+            currentNode = currentNode.next;
+            index--;
+        }
+        return currentNode.value;
+    }
+
+
+
+    //Checking value methods
+
+    isHead(node) {
```

Comment:
```
get isHead() {
   return this.head === node;
}
```
this.isHead

Reply:
this method is not for getting head value but to check that current node is head or not

Reply:
right